core/src/oauth/connection.rs

```diff
@@ -294,6 +302,10 @@ impl ProviderError {
                 code: ConnectionErrorCode::InternalError,
                 message: "Failed to refresh access token.".to_string(),
             },
+            ProviderError::InvalidMetadataError(_) => ConnectionError {
+                code: ConnectionErrorCode::InternalError,
```

Comment:
Arguably that's not an InternalError. You want to add a new ConnectionErrorCode::InvalidMetadataError

Reply:
And route it as BAD_REQUEST in both finalize and access_token